### PR TITLE
Add Support for P2PKH and P2SH-P2WPKH Address Types

### PR DESCRIPTION
**Pull request by notTanveer**
This PR adds support for testing P2PKH and P2SH-P2WPKH address types in the indexer, ensuring silent blocks are correctly processed and retrieved for these transaction types. The test suite has been refactored improving maintainability and extending compatibility with diverse Bitcoin address types.

### CODE REVIEW

File: e2e/helpers/wallet.helper.ts [hunk 1]
```diff
@@ -65,7 +67,12 @@ export class WalletHelper {
         return blockhash;
     }
 
-    async addAmountToAddress(payment: Payment, amount): Promise<UTXO> {
+    async addAmountToAddress(
+        payment: Payment,
+        amount: number,
+        addressType: 'p2wpkh' | 'p2tr' | 'p2pkh' | 'p2sh-p2wpkh',
```
Comment (aruokhai) [inline e2e/helpers/wallet.helper.ts:73]: Kindly create a new type for this,  type AddressType = 'p2wpkh' | 'p2tr' | 'p2pkh' | 'p2sh-p2wpkh';

File: e2e/indexer.e2e-spec.ts [hunk 2]
```diff
@@ -26,45 +26,58 @@ describe('Indexer', () => {
         await shutdownDep();
     });
 
-    it('p2wpkh - should ensure that the correct silent block is fetched', async () => {
-        const taprootOutput = walletHelper.generateAddresses(1, 'p2tr')[0];
-        const p2wkhOutputs = walletHelper.generateAddresses(6, 'p2wpkh');
-        const utxos: UTXO[] = [];
+    async function testSilentBlock(
+        addressType: 'p2wpkh' | 'p2pkh' | 'p2sh-p2wpkh',
```
Comment (aruokhai) [inline e2e/indexer.e2e-spec.ts]: also replace the type

File: e2e/indexer.e2e-spec.ts [hunk 3]
```diff
@@ -26,45 +26,58 @@ describe('Indexer', () => {
...
-            {
-                responseType: 'arraybuffer',
-            },
-        );
+            expect(response.data as Buffer).toEqual(silentBlock);
+        });
+    }
 
-        expect(response.data as Buffer).toEqual(silentBlock);
-    });
```
Comment (aruokhai) [inline e2e/indexer.e2e-spec.ts]: I would prefer it if the test function, in which i mean` it(`${addressType} - should ensure that the correct silent block is fetched`` is not enclosed in a normal function. It makes the code more readable.

File: e2e/indexer.e2e-spec.ts [hunk 4]
```diff
@@ -26,45 +26,58 @@ describe('Indexer', () => {
...
+    testSilentBlock('p2wpkh');
+    testSilentBlock('p2sh-p2wpkh');
+    testSilentBlock('p2pkh');
```
Comment (aruokhai) [inline e2e/indexer.e2e-spec.ts]: Also ensure to add _Pay-to-Taproot (P2TR)_ input as well.

File: e2e/indexer.e2e-spec.ts [hunk 5]
```diff
@@ -26,9 +26,13 @@ describe('Indexer', () => {
...
     });
 
-    async function testSilentBlock(
-        addressType: 'p2wpkh' | 'p2pkh' | 'p2sh-p2wpkh',
-    ) {
+    const addressTypes: ('p2wpkh' | 'p2sh-p2wpkh' | 'p2pkh')[] = [
+        'p2wpkh',
+        'p2sh-p2wpkh',
+        'p2pkh',
+    ];
```
Comment (aruokhai) [inline e2e/indexer.e2e-spec.ts:33]: Also ensure to add Pay-to-Taproot (P2TR) input as well.

**Review by aruokhai — CHANGES_REQUESTED** [review]
Looks good, apart from some minor changes required.

File: e2e/helpers/wallet.helper.ts [hunk 6]
```diff
@@ -19,20 +21,95 @@ export type UTXO = {
...
+            }
+        }
+
+        throw new Error('cant find transaction');
     }
 
-    generateAddresses(count: number, type: 'p2wpkh' | 'p2tr'): Payment[] {
+    generateAddresses(
+        count: number,
+        type: 'p2wpkh' | 'p2tr' | 'p2pkh' | 'p2sh-p2wpkh',
```
Comment (aruokhai) [inline e2e/helpers/wallet.helper.ts:108]: same here

File: e2e/helpers/wallet.helper.ts [hunk 7]
```diff
@@ -21,6 +21,8 @@ export type UTXO = {
     vout: number;
     value: number;
     rawTx: string;
+    addressType: 'p2wpkh' | 'p2tr' | 'p2pkh' | 'p2sh-p2wpkh';
```
Comment (theanmolsharma) [inline e2e/helpers/wallet.helper.ts:24]: please make an enum for this

**Review by theanmolsharma — COMMENTED** [review]
Apart from removing hardcoded enum values for types, I don't have any other comments as of now.

File: e2e/helpers/wallet.helper.ts [hunk 8]
```diff
@@ -7,47 +7,147 @@ import {
...
     Payment,
     Transaction,
+    crypto,
 } from 'bitcoinjs-lib';
 import { btcToSats } from '@e2e/helpers/common.helper';
 import { randomBytes } from 'crypto';
 import { toXOnly } from 'bitcoinjs-lib/src/psbt/bip371';
+import { BitcoinRPCUtil } from '@e2e/helpers/rpc.helper';
+import { assert } from 'console';
+import * as ecpair from 'ecpair';
```
Comment (theanmolsharma) [inline e2e/helpers/wallet.helper.ts:17]:
```suggestion
import { ECPairFactory } from 'ecpair';
```

File: e2e/helpers/wallet.helper.ts [hunk 9]
```diff
@@ -7,20 +7,32 @@ import {
...
 
 initEccLib(ecc);
+const ECPair = ecpair.ECPairFactory(ecc);
+
+export enum AddressType {
+    P2WPKH = 'p2wpkh',
+    P2TR = 'p2tr',
+    P2PKH = 'p2pkh',
+    P2SH_P2WPKH = 'p2sh-p2wpkh',
```
Comment (theanmolsharma) [inline e2e/helpers/wallet.helper.ts:26]:
```suggestion
    P2WPKH = 'P2WPKH',
    P2TR = 'P2TR',
    P2PKH = 'P2PKH',
    P2SH_P2WPKH = 'P2SH_P2WPKH',
```

File: e2e/indexer.e2e-spec.ts [hunk 10]
```diff
@@ -26,45 +26,62 @@ describe('Indexer', () => {
...
+        AddressType.P2PKH,
+        AddressType.P2TR,
+    ];
 
-        for (const output of p2wkhOutputs) {
-            const utxo = await walletHelper.addAmountToAddress(output, 1);
-            utxos.push(utxo);
-        }
+    addressTypes.forEach((addressType) => {
+        it(`${addressType} - should ensure that the correct silent block is fetched`, async () => {
```
Comment (theanmolsharma) [inline e2e/indexer.e2e-spec.ts:37]: instead of this, the cleaner solution would be to `it.each()`

**Review by theanmolsharma — CHANGES_REQUESTED** [review]

**Comment by theanmolsharma** [comment]
Why did you close this?

**Comment by notTanveer** [comment]
> Why did you close this?

merge conflicts

**Pull request closed without merging**